picture-marker-symbols/src/main/java/com/esri/arcgisruntime/sample/picturemarkersymbols/MainActivity.java

```diff
@@ -114,75 +130,122 @@ public void run() {
+    //first, check if there is no sdcard
     if (!Environment.getExternalStorageState().equals(Environment.MEDIA_MOUNTED)) {
-      return false;
+      // no mounted disc, cannot proceed
+      return;
+    }
+
+    //Check for required permission of saving to disc, for devices < android 6 this is set in the manifest and should be granted
+    if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        != PackageManager.PERMISSION_GRANTED) {
+      //no permission, need to task, onRequestPermissionsResult will handle the result
+      ActivityCompat.requestPermissions(this, new String[] { Manifest.permission.WRITE_EXTERNAL_STORAGE },
+          MY_PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE);
     } else {
+      //permission granted, proceed
+      //save the orange marker app resource to disk
+      if (saveFileToExternalStorage()) {
+        createPictureMarkerSymbolFromFile();
+      }
+    }
+  }
+
+  @Override
+  public void onRequestPermissionsResult(int requestCode,
+      String permissions[], int[] grantResults) {
 
-      //build paths
-      mArcGISTempFolderPath = Environment.getExternalStorageDirectory() + File.separator + this.getResources()
-          .getString(R.string.pin_blank_orange_folder_name);
-      mPinBlankOrangeFilePath =
-          mArcGISTempFolderPath + File.separator + this.getResources().getString(R.string.pin_blank_orange_file_name);
+    switch (requestCode) {
+      case MY_PERMISSIONS_REQUEST_WRITE_EXTERNAL_STORAGE: {
+        //If request is cancelled, the result arrays are empty.
+        if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+          //permission granted
+          if (saveFileToExternalStorage()) {
+            createPictureMarkerSymbolFromFile();
+          }
+        }
+      }
+    }
+  }
 
-      //get drawable resource
-      Bitmap bm = BitmapFactory.decodeResource(getResources(), R.drawable.pin_blank_orange);
+  private boolean saveFileToExternalStorage() {
 
-      //create new ArcGIS temp folder
-      File folder = new File(mArcGISTempFolderPath);
-      folder.mkdirs();
+    //build paths
+    mArcGISTempFolderPath = Environment.getExternalStorageDirectory() + File.separator + this.getResources()
+        .getString(R.string.pin_blank_orange_folder_name);
+    mPinBlankOrangeFilePath =
+        mArcGISTempFolderPath + File.separator + this.getResources().getString(R.string.pin_blank_orange_file_name);
 
-      //create file on disk
-      File file = new File(mPinBlankOrangeFilePath);
+    //get drawable resource
+    Bitmap bm = BitmapFactory.decodeResource(getResources(), R.drawable.pin_blank_orange);
 
-      try {
-        OutputStream outStream = new FileOutputStream(file);
-        bm.compress(Bitmap.CompressFormat.PNG, 100, outStream);
-        outStream.flush();
-        outStream.close();
+    //create new ArcGIS temp folder
+    File folder = new File(mArcGISTempFolderPath);
+    folder.mkdirs();
```

Comment:
Result is being ignored, you should test the result and log if successful and send toast message if fails.

You should do similar in `onDestroy` and log result.